collator/src/collator/do_collate/finalize.rs

```diff
@@ -100,7 +102,7 @@ impl Phase<FinalizeState> {
         let mut other_updated_top_shard_diffs_info = FastHashMap::default();
 
         for top_block_id in top_other_updated_shard_blocks_ids.iter() {
-            if top_block_id.seqno == 0 {
+            if top_block_id.seqno == zerostate_id.seqno {
```

Comment:
`top_block_id` be a shard block, so I'm not sure if this check is correct and what was the original idea.

I guess we should skip based on master ref seqno.

@SmaGMan please take a look